pydantic/funcs.py

```diff
@@ -0,0 +1,226 @@
+        if self._schema is None:
+            self._schema = SchemaSerializer(self.core_schema, self.core_config)
+        return self._schema
+
+    def validate_python(self, __object: Any) -> T:
+        return self.validator.validate_python(__object)
+
+    def validate_json(self, __data: str | bytes) -> T:
+        return self.validator.validate_json(__data)
+
+    def dump_python(
+        self,
+        __instance: T,
+        *,
+        mode: Literal['json', 'python'] = 'python',
+        include: IncEx | None = None,
+        exclude: IncEx | None = None,
+        by_alias: bool = False,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        round_trip: bool = False,
+        warnings: bool = True,
+    ) -> Any:
+        return self.serializer.to_python(
+            __instance,
+            mode=mode,
+            by_alias=by_alias,
+            include=include,
+            exclude=exclude,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+            round_trip=round_trip,
+            warnings=warnings,
+        )
+
+    def dump_json(
+        self,
+        __instance: T,
+        *,
+        indent: int | None = None,
+        include: IncEx | None = None,
+        exclude: IncEx | None = None,
+        by_alias: bool = False,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+        round_trip: bool = False,
+        warnings: bool = True,
+    ) -> bytes:
+        return self.serializer.to_json(
+            __instance,
+            indent=indent,
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+            round_trip=round_trip,
+            warnings=warnings,
+        )
+
+
+@overload
+def analyze_type(__type: type[T], *, config: CoreConfig | None = None, parent_depth: int = 2) -> AnalyzedType[T]:
+    ...
+
+
+@overload
+def analyze_type(__type: Any, *, config: CoreConfig | None = None, parent_depth: int = 2) -> AnalyzedType[Any]:
+    ...
+
+
+def analyze_type(__type: Any, *, config: CoreConfig | None = None, parent_depth: int = 2) -> AnalyzedType[Any]:
+    """Generate a pydantic-core schema and other metadata
+    that can be fed into other functions or Validator.
+
+    >>> from typing import List
+    >>> from pydantic import funcs
+    >>> schema = funcs.analyze_type(List[int])
+    >>> assert validate(schema, ['123']) == [123]
+
+    Args:
+        __type: The type to generate a core schema for.
+        config: The pydantic-core config. Defaults to None.
+
+    Returns:
+        core_schema.CoreSchema: A pydantic-core CoreSchema.
+    """
+    # if this is a model with an existing schema short circuit
```

Comment:
Have you thought about putting this logic in the `AnalyzedType.__init__` so you can just do `AnalyzedType(Dict[str, MyModel])`? I'm fine as is if you prefer it, I just imagine it could reduce the number of imports if you want to use `AnalyzedType` as an annotation for a custom function but can't instantiate it without the `analyze_type` function.

Reply:
I have two reasons for this:
1. It’s easier to do overloads on functions because init doesn’t return anything
2. I want us to be able to construct an AnalyzedType with zero cost if we already have the core schema.

Reply:
But what I can do is make this a static method on AnalyzeType

Reply:
both good points